### PR TITLE
Code refactoring

### PR DESCRIPTION
**Pull request by dlasecki**
### Summary

Fixed certain things that SonarLint complained about. They are pretty minor improvements.

### Details and comments

Not sure if pulling out certain strings as constants is a good step... Perhaps we could name them better?

### CODE REVIEW

File: qiskit_nature/mappers/second_quantization/linear_mapper.py [hunk 1]
```diff
@@ -46,17 +46,17 @@ def map(self, second_q_op: SpinOp) -> PauliSumOp:
 
                 if n_x > 0:
                     operator_on_spin_i.append(
-                        reduce(operator.matmul, [spinx for i in range(int(n_x))])
+                        reduce(operator.matmul, [spinx for _ in range(int(n_x))])
```
Comment (mrossinek) [inline qiskit_nature/mappers/second_quantization/linear_mapper.py:49]: Isn't this just `[spinx] * int(n_x)`?
Same for the others..

**Review by mrossinek — CHANGES_REQUESTED** [review]
As far as I know we may *not* change anything in the `gauopen` module. That is also the reason why it get's excluded from all linting jobs. Please undo those changes.

Other than that just a minor comment, thanks @dlasecki :+1:

File: qiskit_nature/mappers/second_quantization/linear_mapper.py (hunk 1 above)
Comment (dlasecki) [reply qiskit_nature/mappers/second_quantization/linear_mapper.py]: Yes, good catch.

**Comment by manoelmarques** [comment]
Yes, the folder "gauopen" has source code from Gaussian that we don't touch. I even have it not checked by style, pylint, mypy etc.

**Review by mrossinek — APPROVED** [review]

**Comment by manoelmarques** [comment]
I added the "backport" status because even though it is not a bug fix, having those changes in stable will reduce the possibility of merge conflicts in future backports.

**Review by pbark — APPROVED** [review]

**Comment by woodsp-ibm** [comment]
This should NOT have been labelled backport. Backporting to stable branch is just for bug fixes deemed critical enough to be part of the next point release.

**Pull request merged**